Tests/tests/test_FrontPanel.cpp

```diff
@@ -0,0 +1,820 @@
+        dispatcher->Release();
+
+        PluginHost::IFactories::Assign(nullptr);
+    }
+};
+
+class FrontPanelInitializedEventDsTest : public FrontPanelInitializedEventTest {
+protected:
+
+    ColorMock colorImplMock;
+
+    FrontPanelInitializedEventDsTest()
+        : FrontPanelInitializedEventTest()
+    {
+        device::FrontPanelIndicator::Color::getInstance().impl = &colorImplMock;   
+    }
+
+    virtual ~FrontPanelInitializedEventDsTest() override
+    {
+        device::FrontPanelIndicator::Color::getInstance().impl = nullptr;
+    }
+};
+
+TEST_F(FrontPanelTest, RegisteredMethods)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("powerLedOn")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("powerLedOff")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setClockBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getClockBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getFrontPanelLights")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getPreferences")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setPreferences")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setLED")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setBlink")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("set24HourClock")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("is24HourClock")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setClockTestPattern")));
+}
+
+TEST_F(FrontPanelInitializedEventDsTest, setBrightnessWIndex)
+{
+    //device::FrontPanelIndicator panel;
+    testing::NiceMock<FrontPanelIndicatorMock> indicatorMock;
+    device::FrontPanelIndicator::getInstance().impl = &indicatorMock;
+
+
+    ON_CALL(indicatorMock, getInstanceString)
+        .WillByDefault(::testing::Invoke(
+            [&](const std::string& name) -> device::FrontPanelIndicator& {
+                EXPECT_EQ("Power", name);
+                return device::FrontPanelIndicator::getInstance();
+            }));
+
+        
+    ON_CALL(frontPanelIndicatorImplMock, getName())
+        .WillByDefault(::testing::Return("Power"));
+
+
+    EXPECT_CALL(indicatorMock, setBrightness(::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](int brightness, bool toPersist) {
+                EXPECT_EQ(brightness, 1);
+            }));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setBrightness"), _T("{\"brightness\": 1,\"index\": \"power_led\"}"), response));
+    EXPECT_EQ(response, string("{\"success\":true}"));
+}
+TEST_F(FrontPanelInitializedEventDsTest, setBrightnessClock)
+{
+
+    testing::NiceMock<FrontPanelIndicatorMock> indicatorMock;
+    device::FrontPanelIndicator::getInstance().impl = &indicatorMock;
+
+    FrontPanelIndicatorMock frontPanelIndicatorMock;
```

Comment:
there's frontPanelIndicatorMock, indicatorMock, frontPanelIndicatorImplMock
these names don't suggest anything
indicatorList suggests it's a list, but it's not
it takes time to figure out your test which makes it not maintainable
please rename, restructure it to be more clear

Reply:
Updated to make it more concise and more consistent with the mock that is being used.